completed-components/exercise-3-html/ProductHeader.tsx

```diff
@@ -18,41 +18,57 @@ const Banner = ({ shouldAnimate = false }: BannerProps) => {
 		}
 	};
 	const incrementSlide = () => {
-		if (currentSlideNum <= slideCount) {
+		if (currentSlideNum < slideCount) {
 			changeSlideNum(currentSlideNum + 1);
 		} else {
 			changeSlideNum(1);
 		}
 	};
+
 	const setInert = (node, slideNumber) => {
 		// workaround for React not recognizing inert
 		// https://github.com/facebook/react/pull/24730
 		return (
 			node && (currentSlideNum === slideNumber ? node.removeAttribute('inert', '') : node.setAttribute('inert', ''))
 		);
 	};
+
 	useEffect(() => {
 		updateSlidePercentage(((currentSlideNum - 1) / 3) * 100);
 	}, [currentSlideNum]);
+
+	async function animate() {
+		incrementSlide();
+	}
+
+	useEffect(() => {
+		if (shouldAnimate) animate();
+	}, []);
+
+	if (shouldAnimate) {
+		setTimeout(() => {
+			animate();
+		}, 3000);
+	}
```

Comment:
This actually comes from the `components` directory having final-exercise code in it. I will try to address this at some point soon.

Reply:
Ah, just kidding, it's boilerplate code. (I've been working through this course in segments rather than all at once.)